lib/rules/no-unknown-custom-media/__tests__/index.mjs

```diff
@@ -0,0 +1,92 @@
+import { stripIndent } from 'common-tags';
+
+import rule from '../index.mjs';
+const { messages, ruleName } = rule;
+
+testRule({
+	ruleName,
+	config: [true],
+
+	accept: [
+		{
+			code: '@custom-media --foo (max-width: 30em); @media (--foo) {};',
+		},
+		{
+			code: '@CUSTOM-MEDIA --foo (max-width: 30em); @media (--foo) {};',
+		},
+		{
+			code: '@cUStOM-mEDiA --foo (max-width: 30em); @media (--foo) {};',
+		},
+		{
+			code: stripIndent`
+				@custom-media --modern (color), (hover);
+				@media (--modern) and (width > 1024px) {}
+			`,
+		},
+		{
+			code: stripIndent`
+				@custom-media --narrow-window (max-width: 30em);
+				@media (--narrow-window) {}
+				@media (--narrow-window) and (script) {}
+			`,
+		},
+		{
+			code: '@keyframes foo {} a { animation-name: foo; }',
+			description: 'at-rule other than media and custom-media',
+		},
+	],
+
+	reject: [
+		{
+			code: '@custom-media --foo (max-width: 30em); @media screen and (--bar) {};',
+			message: messages.expected('--bar'),
+			line: 1,
+			column: 59,
+			endLine: 1,
+			endColumn: 64,
+		},
+		{
+			code: '@custom-media --foo (max-width: 30em); @media (--bar) {};',
```

Comment:
I can't find a description of "hoisting" in <https://www.w3.org/TR/mediaqueries-5/>, but I'm fine with adding such a reject case because it should not request any logic change.

Reply:
> I suppose that custom media rules must be declared before their usage.

see https://github.com/parcel-bundler/lightningcss/issues/49#issuecomment-1013758907

Reply:
see https://github.com/stylelint/stylelint/pull/7594#pullrequestreview-1988807526